pkg/apis/monitoring/v1/prometheusagent_types.go

```diff
@@ -0,0 +1,91 @@
+// limitations under the License.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	PrometheusAgentsKind   = "PrometheusAgent"
+	PrometheusAgentName    = "prometheusagents"
+	PrometheusAgentKindKey = "prometheusagent"
+)
+
+func (l *PrometheusAgent) GetCommonPrometheusFields() CommonPrometheusFields {
+	return l.Spec.CommonPrometheusFields
+}
+
+func (l *PrometheusAgent) SetCommonPrometheusFields(f CommonPrometheusFields) {
+	l.Spec.CommonPrometheusFields = f
+}
+
+func (l *PrometheusAgent) GetTypeMeta() metav1.TypeMeta {
+	return l.TypeMeta
+}
+
+// +genclient
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:categories="prometheus-operator",shortName="promagent"
+// +kubebuilder:printcolumn:name="Version",type="string",JSONPath=".spec.version",description="The version of Prometheus agent"
+// +kubebuilder:printcolumn:name="Desired",type="integer",JSONPath=".spec.replicas",description="The number of desired replicas"
+// +kubebuilder:printcolumn:name="Ready",type="integer",JSONPath=".status.availableReplicas",description="The number of ready replicas"
+// +kubebuilder:printcolumn:name="Reconciled",type="string",JSONPath=".status.conditions[?(@.type == 'Reconciled')].status"
+// +kubebuilder:printcolumn:name="Available",type="string",JSONPath=".status.conditions[?(@.type == 'Available')].status"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:printcolumn:name="Paused",type="boolean",JSONPath=".status.paused",description="Whether the resource reconciliation is paused or not",priority=1
+// +kubebuilder:subresource:status
+
+// Prometheus defines a Prometheus deployment.
+type PrometheusAgent struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Specification of the desired behavior of the Prometheus agent. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	Spec PrometheusAgentSpec `json:"spec"`
+	// Most recent observed status of the Prometheus cluster. Read-only.
+	// More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	Status PrometheusStatus `json:"status,omitempty"`
+}
+
+// DeepCopyObject implements the runtime.Object interface.
+func (l *PrometheusAgent) DeepCopyObject() runtime.Object {
+	return l.DeepCopy()
+}
+
+// PrometheusList is a list of Prometheuses.
+// +k8s:openapi-gen=true
+type PrometheusAgentList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata
+	// More info: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ListMeta `json:"metadata,omitempty"`
+	// List of Prometheuses
+	Items []*Prometheus `json:"items"`
```

Comment:
Nevermiiiiiiiiiiind, I guess reviewing code with Github's interface does help hahaha

The change below solved the issue
```suggestion
	Items []*PrometheusAgent `json:"items"`
```